airbyte-ci/connectors/auto_merge/src/auto_merge/main.py

```diff
@@ -0,0 +1,178 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+from __future__ import annotations
+
+import logging
+import os
+import time
+from collections.abc import Iterable, Iterator
+from contextlib import contextmanager
+from typing import TYPE_CHECKING
+
+from github import Auth, Github
+
+from .consts import AIRBYTE_REPO, AUTO_MERGE_LABEL, BASE_BRANCH, CONNECTOR_PATH_PREFIXES
+from .env import GITHUB_TOKEN, PRODUCTION
+
+if TYPE_CHECKING:
+    from github.Commit import Commit as GithubCommit
+    from github.File import File as GithubFile
+    from github.PullRequest import PullRequest
+    from github.Repository import Repository as GithubRepo
+
+logging.basicConfig()
+logging.getLogger().setLevel(logging.INFO)
+
+
+@contextmanager
+def github_client() -> Iterator[Github]:
+    client = None
+    try:
+        client = Github(auth=Auth.Token(GITHUB_TOKEN), seconds_between_requests=0)
+        yield client
+    finally:
+        if client:
+            client.close()
+
+
+def check_if_modifies_connectors_only(modified_files: Iterable[GithubFile]) -> bool:
```

Comment:
I like this. Ideally, we'd like to verify that only files in a _single_ connector are modified, but honestly I think this would be too granular to focus on here.